docs/source/kinds_of_types.rst

```diff
@@ -1153,3 +1153,176 @@ demonstrates how to work with coroutines. One version
 `uses async/await <https://github.com/python/mypy/blob/master/test-data/samples/crawl2.py>`_
 and one
 `uses yield from <https://github.com/python/mypy/blob/master/test-data/samples/crawl.py>`_.
+
+TypedDict
+*********
+
+Python programs often use dictionaries with string keys to represent objects.
+Here is a typical example:
+
+.. code-block:: python
+
+   movie = {'name': 'Blade Runner', 'year': 1982}
+
+Only a fixed of set of string keys is expected (``'name'`` and
+``'year'`` above), and each key has an independent value type (``str``
+for ``'name'`` and ``int`` for ``'year'`` above). We've previously
+seen the ``Dict[K, V]`` type, which lets you declare uniform
+dictionary types, where every value has the same type, and arbitrary key
+values are supported. This is clearly not a good fit for
+``movie`` above. Instead, you can use a ``TypedDict`` to give a precise
+type for objects like ``movie``, where the type of each
+dictionary value depends on the key:
+
+.. code-block:: python
+
+   from mypy_extensions import TypedDict
+
+   Movie = TypedDict('Movie', {'name': str, 'year': int})
+
+   movie = {'name': 'Blade Runner', 'year': 1982}  # type: Movie
+
+``Movie`` is a TypedDict type with two items: ``'name'`` (with type ``str``)
+and ``'year'`` (with type ``int``). Note that we used an explicit type
+annotation for the ``movie`` variable. This type annotation is
+important -- without it, mypy will try to infer a regular, uniform
+``Dict`` type for ``movie``, which is not what we want here.
+
+.. note::
+
+  If you pass a TypedDict object as an argument to a function, no type
```

Comment:
I am not sure if it is important for RtD, but you use two leading spaces here instead of three as everywhere else.